File: Problems/Leetcode/3001-4000/3201-3300/3254.cpp
#include <bits/stdc++.h>

using namespace std;
using ll = long long;

static constexpr int MOD = 1e9 + 7;

class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k) {
        int n = nums.size();
        deque<int> q;
        vector<int> ans;
        for (int i = 0; i < n; ++i) {
            while (!q.empty() && nums[q.back()] + 1 != nums[i]) {
                q.pop_back();
            }
            q.emplace_back(i);
            if (q.front() <= i - k) {
                q.pop_front();
            }
            if (i >= k - 1) {
                ans.emplace_back(q.size() == k ? nums[q.back()] : -1);
            }
        }
        return ans;
    }
};

int main() {
    Solution sol;

    return 0;
}

File: Problems/Leetcode/3001-4000/3201-3300/3254.py
from collections import deque
from typing import List


class Solution:
    def resultsArray(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        q = deque()
        ans = []
        for i in range(n):
            while q and nums[q[-1]] + 1 != nums[i]:
                q.pop()
            q.append(i)
            if q[0] <= i - k:
                q.popleft()
            if i >= k - 1:
                ans.append(nums[q[-1]] if len(q) == k else -1)
        return ans
